docs/DesignDocs/SGX_Endorsements_V2.md

```diff
@@ -0,0 +1,222 @@
+SGX Attestation Endorsements V2 Updates
+================
+
+Intel SGX released v2 of the Provision Certification Service for ECDSA Attestation.  The
+following APIs were affected:
+
+1. Get TCB Info.
+2. Get Quoting Enclave Identity.
+
+[API Version 1](https://api.portal.trustedservices.intel.com/documentation#pcs-certificate)
+
+[API Version 2](https://api.portal.trustedservices.intel.com/documentation#pcs-certificate-v2)
+
+This document describes the changes for supporting version 2 of the API as well as keeping
+the existing support for version 1 of the API.  This document assumes the user is familiar with
+the SGX attestation endorsements and how it is used during SGX quote verification.
+
+Motivation
+----------
+
+This change is required for customers who would like to use the v2 API web endpoints. Note that the Azure PCK
+Caching Service and the Azure DCAP Client will also need to be updated and it is outside of the this specification.
+
+User Experience
+---------------
+
+The user experience does not change, it should be seamless to the user.  The caller will be able to identify the version of the data by checking the existing `version` field.
+
+The quote verifier will specify whether to use the v1 or v2 Azure PCK Caching Service web API endpoints.  The
+quote verification logic will be able to determine the version of the API by checking the `version` field in endorsement data.
+
+Specification
+-------------
+
+## Differences between V1 and V2 APIs.
+
+The APIs to get the PCK Certificate and the revocation list are the same between the versions.  The APIs to
+get the TCB Info and get QE Identity are different.  The differences are highlighted below.
+
+### Get TCB Info
+1. tcbInfo:version changed from 1 to 2.
+2. New field tcbInfo:tcbType.
+3. New field tcbInfo:tcbEvaluationDataNumber.
+4. Field tcbInfo:tcbLevels:status was renamed to tcbInfo:tcbLevels:tcbStatus.
+5. New fields in tcbInfo:tcbLevels named tcbDate and advisoryIDs.
+
+### Get QE Identity Info
+1. The 'qeIdentity' block was renamed to 'enclaveIdentity'.
+2. A new field, enclaveIdentity:id was added to differentiate QE Id info from Quoting Enclave (QE) and Quoting Validation Enclave (QVE).
+3. enclaveIdentity:version changed from 1 to 2.
+4. New field enclaveIdentity:tcbEvaluationDataNumber.
+5. qeIdentity:isvsvn moved to enclaveIdentity:tcbLevels:tcb:isvsvn.
+6. New fields in enclaveIdentity:tcbLevels named tcbData, tcbStatus and advisoryIDs.
+
```

Comment:
Do you want to mention QvE identity here as well?

Reply:
Sure.